sky/serve/server/impl.py

```diff
@@ -472,6 +472,13 @@ def update(
     workers: Optional[int] = None,
 ) -> None:
     """Updates an existing service or pool."""
+
+    def _validate_task(task: 'task_lib.Task') -> 'task_lib.Task':
```

Comment:
Naming: It's confusing that this doesn't call `task.validate()`. Maybe just `_apply_admin_policy`?